can/io/logger.py

```diff
@@ -228,7 +228,7 @@ def _get_new_writer(self, filename: StringPathLike) -> FileIOMessageWriter:
             return cast(FileIOMessageWriter, logger)
         else:
             raise Exception(
-                f"The log format \".{filename.split('.')[-1]}\" is not "
+                f"The log format \"{os.path.splitext(filename)[-1]}\" is not "
```

Comment:
I used your recommended error message, but slightly adjusted how the extension is found.

Reply:
This does not work for multiple suffixes like `file.asc.gz`

Reply:
I fixed it with your version.